test/e2e/network/proxy.go

```diff
@@ -254,6 +256,78 @@ var _ = SIGDescribe("Proxy", func() {
 				framework.Failf(strings.Join(errs, "\n"))
 			}
 		})
+
+		// Test layout:
+		// 1. Create a LoadBalancer service with no pods on port 9376
+		// 2. Create a Daemonset running serve_hostname on port 9376
+		//    Daemonset is used to guarantee that pod is running on all nodes, and would be reachable without kube-proxy fix
+		// 3. Make sure that traffic that hits the loadbalancer at port 9376 isn't successful
+		It("Should not expose service ports when there are no pods for a service", func() {
+			svcName := "kube-proxy-test"
+			svcPort := int32(9376)
+			jig := framework.NewServiceTestJig(f.ClientSet, svcName)
+
+			// Create service with no pods
+			svc, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Create(&v1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: svcName,
+				},
+				Spec: v1.ServiceSpec{
+					Type: v1.ServiceTypeLoadBalancer,
+					Ports: []v1.ServicePort{
+						v1.ServicePort{
+							Name: "portname1",
+							Port: svcPort,
+							TargetPort: intstr.FromInt(80),
+						},
+					},
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
please annotate this error.
[1]

Reply:
Done

Reply:
sorry i meant this:
https://onsi.github.io/gomega/
see "Annotating Assertions"

Reply:
whoops, my bad.  Will do.

Reply:
Updated